lib/notifications/notification_reason.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Notifications.NotificationReason do
+  use Ecto.Type
+
+  @type t ::
+          :manpower
+          | :disabled
+          | :diverted
+          | :accident
+          | :other
+          | :adjusted
+          | :operator_error
+          | :traffic
+
+  @valid_reasons [
+    :manpower,
+    :disabled,
+    :diverted,
+    :accident,
+    :other,
+    :adjusted,
+    :operator_error,
+    :traffic
+  ]
+
+  @impl true
+  def type, do: :string
+
+  @impl true
+  def cast(reason) do
+    if reason in @valid_reasons do
+      {:ok, reason}
+    else
+      :error
+    end
+  end
+
+  @impl true
+  def load(reason) do
+    reason_as_atom = String.to_atom(reason)
```

Comment:
`String.to_existing_atom` might be a bit safer, since `String.to_atom` has a potential memory leak due to the fact that atoms are apparently not garbage collected. OTOH this data is coming from our own database, not a random piece of user input or something, so it's probably not that huge an issue.

Reply:
Right, we control both sides here so it should be fine, but it would also be good to get in the habit of using `to_existing_atom` when possible.